### PR TITLE
Fix compiler warnings.

### LINKED ISSUE #993: Fix android/ndk compiler warnings
- [ ] [-Wshift-count-overflow]
- [ ] [-Wtautological-constant-out-of-range-compare]
- [ ] [-Wunused-label]
- [ ] [-Winconsistent-missing-override]

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Fix compiler warnings, Android build.  Fixes #993 .
- Unused 'cleanup' label in generated code: refine code generator to include when it is used
- always true (or false) comparisons (platform dependent!) -- add pragma warning and comment
- unused variable: used in assert only, which is ignored on release -- add pragma warning
- shift overflow: use 1ULL for correct larger size.

## Testing instructions

Android compile  /tools/android-build

## Types of changes

* Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

File: src/Keystore/EncryptionParameters.cpp [hunk 1]
```diff
@@ -28,6 +28,8 @@ static Data computeMAC(Iter begin, Iter end, const Data& key) {
     return Hash::keccak256(data);
 }
 
+#pragma GCC diagnostic ignored "-Wunused-variable"
```
Comment (hewigovens) [inline src/Keystore/EncryptionParameters.cpp:31]: I think we should throw an error instead of assert, if encryption failed, better to stop immediately

Comment (catenocrypt) [reply src/Keystore/EncryptionParameters.cpp:31]: This is in constructor, no return value, only way is to throw exception.  And a negative test case also would be needed for it, I'll check if there is a feasible way to make encryption fail.

Comment (catenocrypt) [reply src/Keystore/EncryptionParameters.cpp:31]: The only way aes_encrypt_key() can fail is with wrong key size, but we use constant 16.  I think asser() only is enough.

**Comment by catenocrypt** [comment]
There is no compile warning for project code in any of the 3 CI builds (latest builds, commit 
8741606).

**Comment by catenocrypt** [comment]
@hewigovens , now that there are warnings, maybe we should turn on warnings-to-errors to enforece no warnings.

**Comment by hewigovens** [comment]
New added warnings or existed? I wonder that’s too strict

**Pull request merged**